bluemira/geometry/tools.py

```diff
@@ -151,22 +157,26 @@ def wrapper(*args, **kwargs):
         try:
             return func(*args, **kwargs)
         except cadapi.FreeCADError:
-            data = _reconstruct_function_call(signature, *args, **kwargs)
-            filename = _make_debug_file(func_name)
-
             # Dump the data in the file
-            try:
-                with open(filename, "w") as file:
-                    json.dump(data, file, indent=4, cls=BluemiraGeoEncoder)
+            if LogLevel(int(get_log_level(as_str=False) * 10)) == LogLevel.DEBUG:
+                data = _reconstruct_function_call(signature, *args, **kwargs)
+                filename = _make_debug_file(func_name)
+                try:
+                    with open(filename, "w") as file:
+                        json.dump(data, file, indent=4, cls=BluemiraGeoEncoder)
+
+                    bluemira_debug(
+                        f"Function call {func_name} failed."
+                        f" Debugging information was saved to: {filename}"
+                    )
+                except Exception:  # noqa: BLE001
+                    bluemira_warn(
+                        "Failed to save the failed geometry operation"
+                        f" {func_name} to JSON."
+                    )
 
-                bluemira_debug(
-                    f"Function call {func_name} failed. Debugging information was saved"
-                    f" to: {filename}"
-                )
-            except Exception:  # noqa: BLE001
-                bluemira_warn(
-                    f"Failed to save the failed geometry operation {func_name} to JSON."
-                )
+            else:
+                bluemira_warn("Set logging level to debug to serialise function call")
```

Comment:
I think this could be clearer

"Set logging level to debug to save geometry used in offset"

Or something slightly more descriptive.